### PR TITLE
Interactive List

### PR DESCRIPTION
**Pull request by tomasesquivelgc**
### New with this branch:

 - Checkboxes now toggle each task completed value.
 - Tasks with a true value can be deleted with the "clear all"  button.
 - Changes are automatically saved in the local storage

### CODE REVIEW

File: src/delete-completed-tasks.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  const newArray = [];
+  for (let i = 0; i < task.length; i += 1) {
+    if (task[i].completed === false) {
+      newArray.push(task[i]);
+    }
+  }
+  task.length = 0;
+  task.push(...newArray);
+};
+
```
Comment (christianbiring1) [inline src/delete-completed-tasks.js:11]: - [ ] You did a great job by implementing the delete all completed feature, however, Kindly make sure you use the `filter` method as specified in the project requirement. In that way, your logic will be much more efficient with a good time complexity.

Hint: 
```jsx
arrayOfTasks.filter(tasks= > tasks !== task)
```

**Review by christianbiring1 — COMMENTED** [review]
Hi @tomasesquivelgc , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 



## Highlights
- [ ] The code base is clean and great on implementing modularity to split the task :+1: 
- [ ] All linters checks are green :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary` :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/delete-completed-tasks.js (hunk 1 above)
Comment (tomasesquivelgc) [reply src/delete-completed-tasks.js:11]: @christianbiring1 hello, thank you for your review. I tried using filter() but it doesn't work as intended.
I tried this but for some reason, some elements in the list are duplicated. Could you help me?

`const deleteAllCompleted = (task) => {
  const newArray = task.filter(taskItem => !taskItem.completed);
  task.splice(0, task.length, ...newArray);
  console.log(task);
};`

Comment (tomasesquivelgc) [reply src/delete-completed-tasks.js:11]: hello @christianbiring1 the filter() method is behaving in unpredictable ways. I cannot use it.

**Review by Munsa1 — APPROVED** [review]
Hi @tomasesquivelgc, 
congratulations-1

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 


## Optional suggestions
N/A


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**